thunder/tests/test_auto_register_torchops.py

```diff
@@ -0,0 +1,187 @@
+from functools import partial
+from unittest.mock import patch
+
+import pytest
+import thunder
+import torch
+
+from thunder.tests.framework import requiresCUDA, TorchExecutor
+from thunder.tests.make_tensor import make_tensor
+from thunder.tests.opinfos import get_opinfo, OpInfo
+from thunder.tests.test_einops import skipIfNoCUDA
+
+
+@skipIfNoCUDA
+@pytest.mark.parametrize("requires_grad", [True, False], ids=("train", "inference"))
+@pytest.mark.parametrize("device,", ["cuda", "cpu"])
+def test_torch_ops_trace(device, requires_grad):
+    from itertools import islice
+
+    import thunder.torch.default_torch_ops as ops
+    from torch.testing._internal.common_methods_invocations import op_db
+    from torch.testing._internal.common_device_type import skipCPUIfNoLapack
+
+    name2func = {}
+    [name2func.setdefault(v.__name__, v) for v in ops.torch_auto_registered_ops[torch]]
+    [name2func.setdefault(f"nn.functional.{v.__name__}", v) for v in ops.torch_auto_registered_ops[torch.nn.functional]]
+    # Use the sample input from torch.xx to test torch.tensor.xx
+    [name2func.setdefault(f"Tensor.{v.__name__}", v) for v in ops.torch_auto_registered_ops[torch.Tensor]]
+    op_infos = [opinfo for opinfo in op_db if opinfo.name in name2func]
+    total = 0
+    cnt = 0
+    suc = 0
+    for op_info in op_infos:
```

Comment:
I was wondering if we can parametrize the test over opinfos. I think it would make the test quite easy to read and we won't need the exception handling as we would have the test per op.

We can just move the code which filters opinfo outside into a function and this function would provide values for parametrization.

```python
opinfos = get_filtered_opinfos_for_test()

@pytest.mark.parametrize("opinfo,", opinfos, ids=list(map(lambda opinfo: opinfo.name), opinfos))
```

Wdyt?

Reply:
Hi @kshitij12345 , thanks, it's updated by using parameterized opinfo